AppController/lib/nginx.rb

```diff
@@ -107,12 +107,41 @@ def self.write_fullproxy_version_config(version_key, http_port, https_port,
 
     secure_handlers = HelperFunctions.get_secure_handlers(version_key)
     parsing_log += "Secure handlers: #{secure_handlers}.\n"
-    always_secure_locations = secure_handlers[:always].map { |handler|
-      HelperFunctions.generate_secure_location_config(handler, https_port)
-    }.join
-    never_secure_locations = secure_handlers[:never].map { |handler|
-      HelperFunctions.generate_secure_location_config(handler, http_port)
-    }.join
+
+    always_secure_locations = ""
+    never_secure_locations = ""
+    
+    http_location_params = \
+        "\n\tproxy_set_header      X-Real-IP $remote_addr;" \
+        "\n\tproxy_set_header      X-Forwarded-For $proxy_add_x_forwarded_for;" \
+        "\n\tproxy_set_header      X-Forwarded-Proto $scheme;" \
+        "\n\tproxy_set_header      X-Forwarded-Ssl $ssl;" \
+        "\n\tproxy_set_header      Host $http_host;\n\tproxy_redirect        off;" \
+        "\n\tproxy_pass            http://gae_ssl_#{version_key};" \
+        "\n\tproxy_connect_timeout 600;\n\tproxy_read_timeout    600;" \
+        "\n\tclient_body_timeout   600;\n\tclient_max_body_size  2G;" \
+        "\n    }\n"
```

Comment:
Do we need to set all of these on each URL?

Reply:
@obino Quite frankly I am not entirely sure, I tried to keep the same configuration without grouping the handlers. I could look more into it and see if we do indeed have to set all of these but right now I just kept the same config.